Lab 1 and Lab 2

DanielWap opened the pull request:
Lab 1 and Lab 2

DanielWap committed:
Lab1_3.2.1_MyAdapter name change to Job2dAdapter

Lab1_3.2.1_MyAdaprer name change to Job2dAdapter

DanielWap committed:
Lab 1 3.2.1 Fixed string in Job2dAdapter

Lab 1 3.2.1 Fixed string in Job2dAdapter

DanielWap committed:
Lab 1 3.2.1 Fixed method to draw inside window

Lab 1 3.2.1 Fixed method to draw inside window

DanielWap committed:
Lab 1 3.2.1 Stop unecessary drawer to pop up

Lab 1 3.2.1 Stop unecessary drawer to pop up

DanielWap committed:
Lab 1 3.2.1 Fixed set position in Job2dAda[pter

Lab 1 3.2.1 Fixed set position in Job2dAda[pter

DanielWap committed:
Lab 3.2.3 Add Figure Joe 2 test  drawing figure

Lab 3.2.3 Add Figure Joe 2 test  drawing figure

DanielWap committed:
Lab 1 3.3.1 LineAdapterDriver class and functionality added

Lab 1 3.3.1 LineAdapterDriver class and functionality added

DanielWap committed:
Lab 1 3.3.3 add possibility to use implemented adapter by GUI

Lab 1 3.3.3 add possibility to use implemented adapter by GUI

DanielWap committed:
Lab 2 3.1.1 Add command hierarchy classes

Lab 2 3.1.1 Add command hierarchy classes

DanielWap committed:
Lab 2 3.1.2 Add Command Object implementation

Lab 2 3.1.2 Add Command Object implementation

DanielWap committed:
Lab 2 3.1.3 Add ComplexCommand class

Lab 2 3.1.3 Add ComplexCommand class

DanielWap committed:
Lab 2 3.1.5 Added FactoryManager and draw Square and Rectangle classes

Lab 2 3.1.5 Added FactoryManager and draw Square and Rectangle classes

DanielWap committed:
Lab 3 Additional improvement - comments in DrawSquare and DrawRectangle

Lab 3 Additional improvement - comments in DrawSquare and DrawRectangle

kowallus commented on src/main/java/edu/kis/powp/jobs2d/commands/OperateToCommand.java:
```diff
@@ -0,0 +1,17 @@
+    private int x;
+    private int y;
+
+    public OperateToCommand(int x, int y){
+        this.x = x;
+        this.y = y;
+    }
+
+    @Override
+    public void execute(Job2dDriver driver) {
+		driver.operateTo(x,y);
+	}
```
Original code used tabs instead of spaces. It is important to keep the original convention. It makes tracking actual changes more difficult.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dAdapter.java:
```diff
@@ -23,15 +21,18 @@ public void setPosition(int x, int y) {
 	@Override
 	public void operateTo(int x, int y) {
+		DrawPanelController controller = DrawerFeature.getDrawerController();
+		//DefaultDrawerFrame.getDefaultDrawerFrame().setVisible(true);
```
In revision repo it is safe to delete code instead of using a comment.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dAdapter.java:
```diff
@@ -4,14 +4,12 @@
- * driver adapter to drawer with several bugs.
- */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class Job2dAdapter extends DrawPanelController implements Job2dDriver {
```
The name is better but it does not indicate the purpose of the adapter and it lacks a driver aspect.
Inheritance is not used (no super drawLine calls). Should be removed!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dAdapter.java:
```diff
@@ -4,14 +4,12 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+//import edu.kis.legacy.drawer.panel.DefaultDrawerFrame;
```
remove unused code!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dAdapter.java:
```diff
@@ -23,15 +21,18 @@ public void setPosition(int x, int y) {
 
 	@Override
 	public void operateTo(int x, int y) {
+		DrawPanelController controller = DrawerFeature.getDrawerController();
```
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public void operateTo(int x, int y) {
+        DrawPanelController controller = DrawerFeature.getDrawerController();
```
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public void operateTo(int x, int y) {
+        DrawPanelController controller = DrawerFeature.getDrawerController();
+        ILine nextLine = LineFactory.getSpecialLine();
```
Hardcoded line type. What about 3.3.2 test? Use dependency injection!

kowallus replied:
Use the solution from LineDriverAdapter class!

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/DrawRectangle.java:
```diff
@@ -0,0 +1,26 @@
+		ArrayList<DriverCommand> list = new ArrayList<>();
+		list.add(new SetPositionCommand(-100, -200)); // Point 1
+		list.add(new OperateToCommand(100, -200)); //Draw line to Point 2
+		list.add(new SetPositionCommand(100, -200)); //  Point 2
```
This and following SetPositionCommands are unnecessary! Remove. OperateToCommand already changed a position of devices head.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/DrawRectangle.java:
```diff
@@ -0,0 +1,26 @@
+	public DriverCommand create() {
+		ArrayList<DriverCommand> list = new ArrayList<>();
+		list.add(new SetPositionCommand(-100, -200)); // Point 1
+		list.add(new OperateToCommand(100, -200)); //Draw line to Point 2
```
Unnecessary comments

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/DrawRectangle.java:
```diff
@@ -0,0 +1,26 @@
+
+	@Override
+	public DriverCommand create() {
+		ArrayList<DriverCommand> list = new ArrayList<>();
```
Declaration should point to the interface (ArrayList -> List)

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/DrawRectangle.java:
```diff
@@ -0,0 +1,26 @@
+import edu.kis.powp.jobs2d.commands.OperateToCommand;
+import edu.kis.powp.jobs2d.commands.SetPositionCommand;
+
+public class DrawRectangle implements FactoryManager{
```
Class doesn't draw, its driver command factory.

kowallus replied:
Fix the name

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/FactoryManager.java:
```diff
@@ -0,0 +1,7 @@
+package edu.kis.powp.jobs2d.factories;
+import edu.kis.powp.jobs2d.commands.DriverCommand;
+
+public interface FactoryManager {
```
Why Manager? Rename.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/Job2dDriverTest.java:
```diff
@@ -1,6 +1,6 @@
 package edu.kis.powp.jobs2d;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
+//import edu.kis.powp.jobs2d.Job2dDriver;
```
Remove

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +24,27 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
+		DrawerFeature.getDrawerController().clearPanel();
+
+		switch (e.getActionCommand()){
+			case "Figure Joe 1":
+				DrawerFeature.getDrawerController().clearPanel();
+				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+				break;
+			case "Figure Joe 2":
+				DrawerFeature.getDrawerController().clearPanel();
+				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+				break;
+			case "Square":
+				DrawerFeature.getDrawerController().clearPanel();
+				FactoryManager factory1 = new DrawSquare();
+				factory1.create().execute(driverManager.getCurrentDriver());
+				break;
+			case "Rectangle":
```
Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +24,27 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
+				break;
+			case "Square":
+				DrawerFeature.getDrawerController().clearPanel();
+				FactoryManager factory1 = new DrawSquare();
```
The names factory1 or factory2 should be more precise.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +24,27 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		DrawerFeature.getDrawerController().clearPanel();
```
Remove this line. Panel can be cleared on demand

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +24,27 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
+				break;
+			case "Figure Joe 2":
+				DrawerFeature.getDrawerController().clearPanel();
+				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
```
driverManager.getCurrentDriver() duplicated!

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +24,27 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
+				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+				break;
+			case "Figure Joe 2":
+				DrawerFeature.getDrawerController().clearPanel();
```
Duplicated and unnecessary!

kowallus commented on src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java:
```diff
@@ -54,9 +60,10 @@ private static void setupDefaultDrawerVisibilityManagement(Application applicati
-		defaultDrawerWindow.setVisible(true);
-	}
 
+		//defaultDrawerWindow.setVisible(true); // Uncomment for additional drawer window
+		defaultDrawerWindow.setVisible(false);
```
Remove all comments (in separate commit!)

kowallus commented on src/main/java/edu/kis/powp/jobs2d/factories/DrawSquare.java:
```diff
@@ -0,0 +1,29 @@
+		ArrayList<DriverCommand> list = new ArrayList<>();
+		list.add(new SetPositionCommand(-50, -50)); // Point 1
+		list.add(new OperateToCommand(50, -50)); //Draw line to Point 2
+		list.add(new SetPositionCommand(50, -50)); // Point 2
```
Similar issues

kowallus reviewed (changes requested):
Please improve, clean code: naming, duplications, etc.

DanielWap committed:
Lab 1 and 2 fixes

Fixes for Lab 1 and 2.
The only problem I have was with spaces, it seems that VS Code has troubles with tabs.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java:
```diff
@@ -43,7 +43,9 @@ private static void setupDrivers(Application application) {
+		Job2dDriver testDriver = new MyAdapter(drawPanelController);
 		DriverFeature.addDriver("Job 2d Adapter", testDriver);
 
 		Job2dDriver lineDrawerDriver = new LineDrawerAdapter();
```
I think that this line doesn't work.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java:
```diff
@@ -0,0 +1,49 @@
+	@Override
+	public void operateTo(int x, int y) {
+		int distance = (int) Math.sqrt(Math.pow(this.startX -x, 2)+Math.pow(this.startY -y, 2));
+		logger.info("total distance: " + (LineDriverAdapter.totalDistance += distance));
```
unnecessary code (from project?)

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * Line adapter - Job2dDriver with DrawPanelController object.
+ */
+public class LineDriverAdapter implements Job2dDriver {
```
This class seems not to be used.

kowallus reviewed (changes requested):
I have done a small check. Still many issues are left to be fixed

kowallus commented:
OK. 3 pts
